refactor: add cppcodec for checksum and base64 logic

dudantas opened the pull request:
**Summary**
Replace custom Adler-32 and Base64 with zlib and cppcodec. Make XOR in-place. Centralize `<zlib.h>` in PCH. Add `cppcodec` to the build.

---

## Description

* **Checksum:** add `stdext::computeChecksum(std::span<const uint8_t>)` using `::adler32`; remove old Adler-32.

  * Update `InputMessage::readChecksum` and `OutputMessage::writeChecksum` to call it.
* **Base64:** switch to `cppcodec::base64_rfc4648` for encode/decode. Graceful handling of invalid input.
* **XOR:** change `Crypt::xorCrypt` to an in-place mutation to avoid copies.
* **Includes:** move `<zlib.h>` to `src/framework/pch.h`; drop local includes.
* **Build:** add `cppcodec` to `vcpkg.json`; in CMake use `find_package(cppcodec CONFIG REQUIRED)` and link `cppcodec::cppcodec`.

---

## Behavior

### Actual

* Project uses bespoke Adler-32 and Base64.
* XOR returns a new string.
* `<zlib.h>` included in multiple files.

### Expected

* Single checksum path via zlib in `stdext::computeChecksum`.
* Base64 via a maintained library.
* XOR is in-place.
* `<zlib.h>` only in the PCH.

---

## Fixes

N/A

---

## Type of change

* [x] Bug fix (non-breaking change which improves correctness and maintenance)
* [x] This change requires a documentation update (build deps: add `cppcodec`)

---

## Checklist

* [x] Code follows project style
* [x] Self-review done
* [x] PR checks reviewed
* [x] Comments added where needed
* [x] Documentation updated for new dependency
* [x] No new warnings

dudantas committed:
Refactor checksum and base64 logic, add cppcodec

Replaces custom Adler-32 checksum implementation with zlib's adler32 via a new stdext::computeChecksum function. Refactors base64 encode/decode to use cppcodec, updates XOR crypt to operate in-place, and move zlib includes from several files to the pch.h. Updates CMake and vcpkg to include cppcodec as a dependency.

dudantas committed:
fix: compute checksum linkage

kokekanon commented:
compilation error vs comunity 2022 vs solution

```log
1>D:\github\otclient.readme\src\framework\stdext\math.cpp(35,35): error C2039: 'span': is not a member of 'std'
1>(compiling source file 'otclient/x64/OpenGL/unity_WOJJ61WZSWNDGS4G.cpp')
1>D:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.39.33519\include\random(33,1):
1>see declaration of 'std'
1>D:\github\otclient.readme\src\framework\stdext\math.cpp(35,35): error C7568: argument list missing after assumed function template 'span'
1>(compiling source file 'otclient/x64/OpenGL/unity_WOJJ61WZSWNDGS4G.cpp')
1>D:\github\otclient.readme\src\framework\stdext\math.cpp(35,35): error C2062: type 'unknown-type' unexpected
1>(compiling source file 'otclient/x64/OpenGL/unity_WOJJ61WZSWNDGS4G.cpp')
1>D:\github\otclient.readme\src\framework\stdext\math.cpp(36,5): error C2143: syntax error: missing ';' before '{'
1>(compiling source file 'otclient/x64/OpenGL/unity_WOJJ61WZSWNDGS4G.cpp')
1>D:\github\otclient.readme\src\framework\stdext\math.cpp(36,5): error C2447: '{': missing function header (old-style formal list?)
1>(compiling source file 'otclient/x64/OpenGL/unity_WOJJ61WZSWNDGS4G.cpp')
1>unity_AKX6MGN260LFIYA3.cpp
```

## posible fix: 
https://github.com/mehah/otclient/blob/d08fce484c5d7fa4d444e9ef665824da55cdcc5a/src/framework/stdext/math.cpp#L23-L28

add
```cpp
#include <span>
#include <zlib.h>
```

dudantas commented:
> compilation error vs comunity 2022 vs solution
> 
> ```
> 1>D:\github\otclient.readme\src\framework\stdext\math.cpp(35,35): error C2039: 'span': is not a member of 'std'
> 1>(compiling source file 'otclient/x64/OpenGL/unity_WOJJ61WZSWNDGS4G.cpp')
> 1>D:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.39.33519\include\random(33,1):
> 1>see declaration of 'std'
> 1>D:\github\otclient.readme\src\framework\stdext\math.cpp(35,35): error C7568: argument list missing after assumed function template 'span'
> 1>(compiling source file 'otclient/x64/OpenGL/unity_WOJJ61WZSWNDGS4G.cpp')
> 1>D:\github\otclient.readme\src\framework\stdext\math.cpp(35,35): error C2062: type 'unknown-type' unexpected
> 1>(compiling source file 'otclient/x64/OpenGL/unity_WOJJ61WZSWNDGS4G.cpp')
> 1>D:\github\otclient.readme\src\framework\stdext\math.cpp(36,5): error C2143: syntax error: missing ';' before '{'
> 1>(compiling source file 'otclient/x64/OpenGL/unity_WOJJ61WZSWNDGS4G.cpp')
> 1>D:\github\otclient.readme\src\framework\stdext\math.cpp(36,5): error C2447: '{': missing function header (old-style formal list?)
> 1>(compiling source file 'otclient/x64/OpenGL/unity_WOJJ61WZSWNDGS4G.cpp')
> 1>unity_AKX6MGN260LFIYA3.cpp
> ```
> 
> ## posible fix:
> https://github.com/mehah/otclient/blob/d08fce484c5d7fa4d444e9ef665824da55cdcc5a/src/framework/stdext/math.cpp#L23-L28
> 
> add
> 
> ```c++
> #include <span>
> #include <zlib.h>
> ```

Apparently, it's a problem with the solution not correctly reading pch.h

dudantas committed:
Enable precompiled headers in Visual Studio project

Added pch.cpp to the project and configured precompiled header usage in settings.props. This improves build performance by using framework/pch.h as the precompiled header and forcing its inclusion in compilation.

dudantas committed:
Merge branch 'main' into dudantas/change-manual-base64-to-automatic

dudantas committed:
Unify unsigned long Lua value casting

Replaces platform-specific ulong handling with a typedef for unsigned long, improving portability and reducing conditional compilation in Lua value casting functions.

dudantas committed:
Refactor unsigned long casting for Lua values

Unified handling of 64-bit unsigned types by introducing lua_u64 and using std::conditional_t for platform compatibility. Simplified push and cast functions for unsigned long and uint64_t, ensuring consistent 64-bit behavior.

dudantas committed:
Add Lua value cast and push for unsigned long

Implements push_luavalue and luavalue_cast overloads for unsigned long types, handling both 32-bit and 64-bit cases. This improves type support for Lua value conversions in the engine.

dudantas committed:
Refactor lua_u64 handling in luavaluecasts.h

Moved lua_u64 type definitions and static_assert above related functions for clarity. Updated push_luavalue and luavalue_cast for lua_u64 to use SFINAE, ensuring correct overloads when unsigned long is not 64-bit.

dudantas committed:
Refactor lua_u64 value casting and pushing functions

Replaces SFINAE-based overloads for push_luavalue and luavalue_cast with template functions using std::enable_if and std::is_same. This improves type safety and clarity for handling lua_u64 types.

dudantas committed:
Add pch.cpp and remove Visual Studio filters file

Added src/framework/pch.cpp for precompiled header implementation. Removed vc17/otclient.vcxproj.filters to clean up project file organization.

dudantas committed:
Add '*.filters' to .gitignore

Added '*.filters' to the .gitignore file to prevent Visual Studio filter files from being tracked in version control.

dudantas committed:
Update pch.cpp path in project file

Changed the path of pch.cpp in otclient.vcxproj to reference the file in the src/framework directory. This ensures the correct file is included during compilation.

dudantas commented:
@kokekanon, can you please recheck this? I've added the pch.cpp to the solution.

kokekanon commented:
@dudantas I confirm compiles in vs solution 2022
```cpp
1>unity_LYKZE09832UFYFGY.cpp
1>unity_VTY1U328R6T5J0CN.cpp
1>unity_HP033YRVZXFMW2YE.cpp
1>   Creating library D:\github\otclient.readme\vc17\../otclient_gl_x64.lib and object D:\github\otclient.readme\vc17\../otclient_gl_x64.exp
1>Generating code
1>Finished generating code
```

dudantas commented:
> @dudantas I confirm compiles in vs solution 2022
> 
> ```c++
> 1>unity_LYKZE09832UFYFGY.cpp
> 1>unity_VTY1U328R6T5J0CN.cpp
> 1>unity_HP033YRVZXFMW2YE.cpp
> 1>   Creating library D:\github\otclient.readme\vc17\../otclient_gl_x64.lib and object D:\github\otclient.readme\vc17\../otclient_gl_x64.exp
> 1>Generating code
> 1>Finished generating code
> ```

Thanks for your feedback.
